### PR TITLE
Added source to .etym , Fixes #343

### LINKED ISSUE #343: Make .etym command append link to full article
Currently, the .etym command outputs an article (or part of an article if its long enough), but does not include a link to the article. For example,

> .etym frog

will output

> frog (n.1) Old English frogga "frog," a diminutive of frosc, forsc, frox "frog," a common Germanic word but with different formations that are difficult to explain (cognates: Old Norse froskr, Middle Dutch vorsc, German Frosch "frog"), probably literally "hopper," from PIE root *preu- "to hop" (source also of Sanskrit provate "hops," Russian prygat "to hop, jump"). Watkins calls the Old English ...

The link to the full article is https://www.etymonline.com/word/frog#etymonline_v_14201

Regardless of article length, the output should be limited to X characters with room for a link to the full article at the end. Would it be possible to modify this behavior? Thanks.

### PR DESCRIPTION
**Pull request by GaeaKat**

### CODE REVIEW

File: plugins/etymology.py [hunk 1]
```diff
@@ -47,7 +49,8 @@ def etymology(text, reply):
 
     etym = ' '.join(etym.split())
 
-    if len(etym) > 400:
-        etym = etym[:etym.rfind(' ', 0, 400)] + ' ...'
+    if len(etym) > 200:
+        etym = etym[:etym.rfind(' ', 0, 200)] + ' ...'
```
Comment (linuxdaemon) [inline plugins/etymology.py]: Please replace this with `truncate()` from `cloudbot.util.formatting`

File: plugins/etymology.py [hunk 2]
```diff
@@ -47,7 +49,8 @@ def etymology(text, reply):
...
+    etym = etym + " Source: " + response.url
```
Comment (linuxdaemon) [inline plugins/etymology.py]:
```suggestion
    etym += " Source: " + response.url
```

File: plugins/etymology.py [hunk 3]
```diff
@@ -28,6 +28,8 @@ def etymology(text, reply):
     try:
         response.raise_for_status()
     except HTTPError as e:
+        if response.status_code == 404:
+            return 'No etymology found for {} :('.format(text)
```
Comment (linuxdaemon) [inline plugins/etymology.py]:
```suggestion
            return "No etymology found for {} :(".format(text)
```
Text presented to the user should be enclosed in double quotes

File: plugins/etymology.py [hunk 4]
```diff
@@ -28,6 +28,8 @@ def etymology(text, reply):
     try:
         response.raise_for_status()
     except HTTPError as e:
+        if response.status_code == 404:
```
Comment (linuxdaemon) [inline plugins/etymology.py]: It may be better to use `e.response` like the `reply()` line below does

**Review by linuxdaemon — CHANGES_REQUESTED** [review]

**Comment by linuxdaemon** [comment]
Please add a `Fixes <issue>` line to the PR description so GitHub will auto close it

**Review by linuxdaemon — APPROVED** [review]

File: plugins/etymology.py [hunk 5]
```diff
@@ -13,7 +13,7 @@
 from requests import HTTPError
 
 from cloudbot import hook
-
+from cloudbot.util import formatting
```
Comment (linuxdaemon) [inline plugins/etymology.py:18]: There should be 2 blank lines before and after a function

**Review by linuxdaemon — CHANGES_REQUESTED** [review]

File: plugins/etymology.py [hunk 6]
```diff
@@ -38,16 +41,14 @@ def etymology(text, reply):
...
 
     etym = ' '.join(etym.splitlines())
 
     etym = ' '.join(etym.split())
 
-    if len(etym) > 400:
-        etym = etym[:etym.rfind(' ', 0, 400)] + ' ...'
+    etym = formatting.truncate(etym, 200)
+
+    etym += " Source: " + response.url
```
Comment (linuxdaemon) [inline plugins/etymology.py]: Wrapping `response.url` in `web.try_shorten()` would make it more consistent with other commands

**Review by linuxdaemon — CHANGES_REQUESTED** [review]

File: plugins/etymology.py (hunk 6 above)
Comment (GaeaKat) [reply plugins/etymology.py]: This should do it
```suggestion
    etym += " Source: " + web.try_shorten(response.url)
```

**Review by linuxdaemon — APPROVED** [review]

**Pull request merged**